### PR TITLE
syncer: support  schema/table sharding

### PR DESCRIPTION
**Pull request by GregoryIan**
* don't support splitting source-schema into multiple target schema
* insert conflict rules would cause error
```
[[route-rules]]
pattern-schema = "route_*"
pattern-table = "abc_*"
target-schema = "route"
target-table = "abc"
```
example above support routing `route_1`.`abc_1` to `route`.`abc` (and `route_1` to `route` implicitly)

### CODE REVIEW

File: syncer/db.go [hunk 1]
```diff
@@ -455,54 +455,121 @@ func resolveDDLSQL(sql string) (sqls []string, ok bool, err error) {
...
+func genTableName(schema string, table string) *TableName {
+	return &TableName{Schema: schema, Name: table}
 
 }
 
-func parserDDLTableName(sql string) (TableName, error) {
+// the result contains [tableName] excepted create table like and rename table
+// for `create table like` DDL, result contains [sourceTableName, sourceRefTableName]
+// for rename table ddl, result contains [targetOldTableName, sourceNewTableName]
+func parserDDLTableNames(sql string) ([]*TableName, error) {
```
Comment (holys) [inline syncer/db.go]: parseDDLTableNams ? not parser

File: syncer/db.go [hunk 2]
```diff
@@ -455,54 +455,121 @@ func resolveDDLSQL(sql string) (sqls []string, ok bool, err error) {
...
 		if len(v.Tables) != 1 {
 			return res, errors.Errorf("drop table with multiple tables, may resovle ddl sql failed")
 		}
-		res = genTableName(v.Tables[0].Schema.L, v.Tables[0].Name.L)
+		res = append(res, genTableName(v.Tables[0].Schema.L, v.Tables[0].Name.L))
+	case *ast.CreateIndexStmt:
+		res = append(res, genTableName(v.Table.Schema.L, v.Table.Name.L))
+	case *ast.DropIndexStmt:
+		res = append(res, genTableName(v.Table.Schema.L, v.Table.Name.L))
+	case *ast.TruncateTableStmt:
```
Comment (holys) [inline syncer/db.go]: Why don't you put table-like statements  together?


```
case *ast.CreateTableStmt
case *ast.DropTableStmt:
case *ast.TruncateTableStmt:
case *ast.AlterTableStmt:
case *ast.RenameTableStmt:
```

**Comment by holys** [comment]
the rest LGTM

File: syncer/db.go (hunk 1 above)
Comment (GregoryIan) [inline syncer/db.go]: use parser to fetch tableNames, personal taste

File: syncer/db.go (hunk 2 above)
Comment (holys) [inline syncer/db.go]: just  the order

Comment (GregoryIan) [inline syncer/db.go]: if so, `v` can't be convert to specified type

Comment (GregoryIan) [inline syncer/db.go]: alright， I would adjust it

**Comment by jomenxiao** [comment]
LGTM

**Pull request merged**